test/utils/ktesting/format.go

```diff
@@ -0,0 +1,49 @@
+/*
+Copyright 2023 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package ktesting
```

Comment:
would we want the example https://github.com/kubernetes/kubernetes/pull/115950 was trying to fix (in kubelet server.go) to reference the test/utils/ktesting package?

Reply:
This is up for debate. I picked `ktesting` because I intend to use YAML here for pretty-printing, if I can convince myself that better formatting during unit testing is really desirable. It could hide that "normal" formatting isn't as good (klog cannot (or should not?) depend on "sigs.k8s.io/yaml").

If "sigs.k8s.io/yaml" was an acceptable dependency for klog, then klog would be a good place for this code.

"staging/src/k8s.io/component-base/logs" might also be okay.

Reply:
Referencing the test package in normal code might not seem like a good idea. 

+1 for `klog` or `component-base/logs`

we can also consider using `gopkg.in/yaml.v2`